src/transport/net_ib.cc

```diff
@@ -1237,11 +1258,23 @@ ncclResult_t ncclIbTest(void* request, int* done, int* sizes) {
     for (int w=0; w<wrDone; w++) {
       struct ibv_wc *wc = wcs+w;
       if (wc->status != IBV_WC_SUCCESS) {
+        // NOTE: this error trace is only printed upon completion failure.
         char line[SOCKET_NAME_MAXLEN+1];
+        char localGid[INET6_ADDRSTRLEN];
+        char remoteGid[INET6_ADDRSTRLEN];
         union ncclSocketAddress addr;
         ncclSocketGetAddr(r->sock, &addr);
-        WARN("NET/IB : Got completion from peer %s with error %d, opcode %d, len %d, vendor err %d",
-             ncclSocketToString(&addr, line), wc->status, wc->opcode, wc->byte_len, wc->vendor_err);
+        if (r->sendComm != NULL) {
+          inet_ntop(AF_INET6, &(r->sendComm->localGid), localGid, sizeof(localGid));
+          inet_ntop(AF_INET6, &(r->sendComm->remoteGid), remoteGid, sizeof(remoteGid));
```

Comment:
These inet_ntop() calls may fail by returning NULL (see https://man7.org/linux/man-pages/man3/inet_ntop.3.html). For completeness, I think we should check inet_ntop() failures, in which case we can either fall back to the old WARN message or just print "unknown" for whichever GID that failed?

Also I think you can refactor this error handle out to a helper method  if you feel it is getting complex.

Reply:
Changed the logic to handle inet_ntop failures.